### PR TITLE
Add type guard to empty file check

### PR DESCRIPTION
**Pull request by solmstead25**
Getting an error when using with another loader that produces `options.data === undefined` under webpack4.

### CODE REVIEW

File: lib/loader.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ function sassLoader(content) {
     ));
 
     // Skip empty files, otherwise it will stop webpack, see issue #21
-    if (options.data.trim() === "") {
+    if (typeof options.data !== "string" || options.data.trim() === "") {
```
Comment (evilebottnawi) [inline lib/loader.js]: Let's add additional comment about why this and we can merge this :+1: (don't remove current comment just add new below)

File: lib/loader.js [hunk 2]
```diff
@@ -63,6 +63,7 @@ function sassLoader(content) {
     ));
 
     // Skip empty files, otherwise it will stop webpack, see issue #21
+    // Guard against possibility of non-string data, see #569
```
Comment (evilebottnawi) [inline lib/loader.js]: `see PR #569`

**Comment by jhnns** [comment]
Thanks for your PR.

Can you describe a situation where this happens? It looks like you're sharing the `options` object between loaders which is not what you should do in the first place. Every loader should get its own options object.

**Comment by solmstead25** [comment]
@jhnns  When using `vue-loader`, it encountered a parsing error in a single file component, which lead to an `undefined` value for `options.data` at that point in execution.

A cast in the `normalizeOptions` may be a better option?

https://github.com/webpack-contrib/sass-loader/blob/437450fb782308270c12a794fe5ca202de2930a7/lib/normalizeOptions.js#L24

Basically when `options.data` is falsey and `content` isn't a string, this will occur.

I didn't dig deep enough to know the scenarios where this happens, and was just trying to help guard against someone having to debug something in the future over a type error.

**Comment by jhnns** [comment]
> When using vue-loader, it encountered a parsing error in a single file component, which lead to an undefined value for options.data at that point in execution.

But that sounds like the error is somewhere in vue-loader? :)

> A cast in the normalizeOptions may be a better option?

I think we should rather throw an error. Just passing an empty string as result doesn't seem like something you'd really want.

> I didn't dig deep enough to know the scenarios where this happens, and was just trying to help guard against someone having to debug something in the future over a type error.

Thanks 👍 :)

**Pull request closed without merging**